src/ansys/acp/core/_tree_objects/interface_layer.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (C) 2022 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from __future__ import annotations
+
+from collections.abc import Iterable
+import dataclasses
+
+import numpy as np
+
+from ansys.api.acp.v0 import interface_layer_pb2, interface_layer_pb2_grpc
+
+from .._utils.property_protocols import ReadWriteProperty
+from ._grpc_helpers.linked_object_list import (
+    define_linked_object_list,
+    define_polymorphic_linked_object_list,
+)
+from ._grpc_helpers.property_helper import (
+    grpc_data_property,
+    grpc_data_property_read_only,
+    mark_grpc_properties,
+)
+from ._mesh_data import (
+    ElementalData,
+    NodalData,
+    ScalarData,
+    VectorData,
+    elemental_data_property,
+    nodal_data_property,
+)
+from .base import CreatableTreeObject, IdTreeObject
+from .element_set import ElementSet
+from .enums import status_type_from_pb
+from .object_registry import register
+from .oriented_selection_set import OrientedSelectionSet
+
+
+@dataclasses.dataclass
+class InterfaceLayerElementalData(ElementalData):
+    """Represents elemental data for a Modeling Ply."""
+
+    normal: VectorData | None = None
+    design_angle: ScalarData[np.float64] | None = None
```

Comment:
@roosre I added the `design_angle` since the mesh query returned a not-`None` value, but doubt whether this actually has meaning or would always be all-zeroes.

Reply:
@greschd Interface Layers neither have an angle, nor a material. They link to an OSS but that is only required for the orientations where the reference directions is not used.
It can well be that the mesh query is not consistent wrt to unavailable data.